File: javascript/quickSort.js
"use strict";

const quickSort = (array) => {
  if (array.length < 2) {
    return array;
  }
  const pivot = array[0];
  const keysAreLessPivot = array.slice(1).filter((key) => key <= pivot);
  const keysAreMorePivot = array.slice(1).filter((key) => key > pivot);
  return [
    ...quickSort(keysAreLessPivot),
    pivot,
    ...quickSort(keysAreMorePivot),
  ];
};

let myArray = [10, 5, 2, 3];
console.log(quickSort(myArray));

File: python/quick_sort.py
def quicksort(array):
    if len(array) < 2:
        return array
    else:
        pivot = array[0]
        less = [i for i in array[1:] if i <= pivot]
        more = [i for i in array[1:] if i > pivot]

        return quicksort(less) + [pivot] + quicksort(more)


my_array = [12, 34, 2, 6, 7, 82]

print(quicksort(my_array))
